### PR TITLE
Fix clippy errors

### PR DESCRIPTION
**Pull request by isabeltomb**

### CODE REVIEW

File: libtransact/src/workload/runner.rs [hunk 1]
```diff
@@ -1002,9 +1002,9 @@ pub struct Link {
 /// Used for deserializing `GET /batch_status` responses.
 #[derive(Debug, Deserialize)]
 struct BatchInfo {
-    pub id: String,
+    pub _id: String,
```
Comment (peterschwarz) [inline libtransact/src/workload/runner.rs:1005]: If these fields are not used, we should remove them, but add a comment with an example of the complete response, as of this commit.  

E.g.
```
/// This struct is a subset of the batch status response.  The complete response looks like
/// {
///     id: "<id string>",
```
etc

This applies to the child structs/enums of this object, as well.

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: libtransact/src/workload/runner.rs (hunk 1 above)
Comment (isabeltomb) [reply libtransact/src/workload/runner.rs]: Did this for `BatchInfo` but `InvalidTransaction` and `ValidTransaction` are printed out in error messages if a batch result doesn't match the expected result, so removing the "unused fields" would remove some important information from those error messages. Implemented display for `InvalidTransaction` and `ValidTransaction` instead.

Comment (peterschwarz) [reply libtransact/src/workload/runner.rs]: Discussing this further, the right fix here would be a) remove the `pub` modifiers from the fields, and then add `#[allow(dead_code)]` to each with a comment of "these are deserialized and available for debugging"

Comment (isabeltomb) [reply libtransact/src/workload/runner.rs]: Fixed

**Review by peterschwarz — APPROVED** [review]

**Review by agunde406 — APPROVED** [review]

**Pull request merged**